weekDownloader.py

```diff
@@ -62,15 +66,51 @@ def main(argv):
     all_units = get_unit_list(get_all_weeks(course_data), token)
     all_steps = get_steps_list(all_units, int(argv[1]), token)
     only_video_steps = get_only_video_steps(all_steps, token)
-    url_list_with_q = [x['video']['urls'][0] for x in only_video_steps]
+
+    url_list_with_q = []
+
+    for video_step in only_video_steps:
+        video_link = None
+
+        # Check a video quality.
+        for url in video_step['video']['urls']:
+            if url['quality'] == argv[3]:
+                video_link = url
+
+        # If the is no required video quality then download
+        # with the best available quality.
+        if video_link is None:
+            print("The requested quality = {} is not available!".format(argv[2]))
+
+            video_link = video_step['video']['urls'][0]
+
+        url_list_with_q.append(video_link)
+
     url_list = [x['url'] for x in url_list_with_q]
+
+    folder_name = argv[2]
+
+    # Create a folder if needed.
+    if not os.path.isdir(folder_name):
+        try:
+            # Create a directory for a particular week in the course.
+            os.makedirs(folder_name)
+        except PermissionError:
+            print("Run the script from admin")
+            exit(1)
+        except FileExistsError:
+            print("Please delete the folder " + folder_name)
+            exit(1)
+
+    print('Folder_name ', folder_name)
+
     for i, el in enumerate(url_list):
-        filename = 'Video_'+str(i)+'.mp4'
+        filename = os.path.join(folder_name, 'Video_' + str(i) + '.mp4')
         print('Downloading file ', filename)
         urllib.request.urlretrieve(el, filename)
```

Comment:
jfyi it seems like for downloading large files it's better to use requests library - http://stackoverflow.com/questions/16694907/how-to-download-large-file-in-python-with-requests-py

I will rewrite this part later